src/TypeSpec.Extension/Emitter.Csharp/src/emitter.ts

```diff
@@ -72,6 +72,14 @@ export async function $onEmit(context: EmitContext<AzureNetEmitterOptions>) {
                 : undefined);
         configurations["enable-internal-raw-data"] =
             options["enable-internal-raw-data"];
+        const examplesDir =
+            options["examples-dir"] ?? options["examples-directory"];
```

Comment:
why is the fallback needed? due to legacy config?

Reply:
yeah TCGC now has two configs  - directory and dir, they want to keep the dir version to align some existing config names, and the directory version will be removed.
But since now we do have two configs, we have to read them both.